src/server/string_family.cc

```diff
@@ -536,6 +547,14 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
       if (int_arg >= kMaxExpireDeadlineSec * 1000) {
         return builder->SendError(InvalidExpireTime("set"));
       }
+
+      // Replicate any expiration argument as PXAT.
+      // TODO: Skip this flow is journal is disabled?
+      {
+        auto abs_time_str = absl::StrCat(abs_int_arg_ms.value_or(TtlToAbsExpiry(int_arg)));
+        cntx->transaction->RewriteAutoJournal({{i - 1, "PXAT"}, {i, abs_time_str}});
+      }
+
```

Comment:
Haven't yet found a way to check from the coordinator that the journal is enabled 🤔